common/motion_utils/src/distance/distance.cpp

```diff
@@ -0,0 +1,264 @@
+// Copyright 2023 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "motion_utils/distance/distance.hpp"
+
+namespace motion_utils
+{
+bool validCheckDecelPlan(
+  const double v_end, const double a_end, const double v_target, const double a_target,
+  const double v_margin, const double a_margin)
+{
+  const double v_min = v_target - std::abs(v_margin);
+  const double v_max = v_target + std::abs(v_margin);
+  const double a_min = a_target - std::abs(a_margin);
+  const double a_max = a_target + std::abs(a_margin);
+
+  if (v_end < v_min || v_max < v_end) {
+    std::cerr << "[validCheckDecelPlan] valid check error! v_target = " << v_target
+              << ", v_end = " << v_end << std::endl;
+    return false;
+  }
+  if (a_end < a_min || a_max < a_end) {
+    std::cerr << "[validCheckDecelPlan] valid check error! a_target = " << a_target
+              << ", a_end = " << a_end << std::endl;
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * @brief update traveling distance, velocity and acceleration under constant jerk.
+ * @param (x) current traveling distance [m/s]
+ * @param (v) current velocity [m/s]
+ * @param (a) current acceleration [m/ss]
+ * @param (j) target jerk [m/sss]
+ * @param (t) time [s]
+ * @return updated traveling distance, velocity and acceleration
+ */
+std::tuple<double, double, double> update(
```

Comment:
[nits]
I think it is better to move this helper function to anonymous namespace.